### DISCUSSION TITLE
Get notifications of Discussions in slack (or equivalent)

### DISCUSSION THREAD

**Discussion by prudhomm** [community/community#2844] [Discussions] [answered]
We use slack as real-time chat. Would it be possible to get the notifications of discussions/answers in slack ?

**Comment by HPaulson** [comment]
@prudhomm This would be available with @becca's idea in #31 regarding webhook messages for discussion events, however as of right now, this isn't possible using just GitHub.

On the other hand, though, Slack and GitHub both have great APIs. Using both of them, you could listen for the discussions event, then post to slack. At the current moment, I don't see a Discussions event on the GitHub [Api Docs](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/github-event-types) just yet, however, I'm sure the team will add it there soon enough.

Hope this helps! :+1:

**Reply by jakeatoms** [reply]
I would imagine this would fall into the same UI as Scheduled Reminders under Settings, which already sends messages to a configured Slack workspace. It includes real-time alerts for things like PR comments.

**Reply by folex** [reply]
Hi @HPaulson! Thanks for the answer. There's still no Discussions event in the GitHub API AFAICT. Is there a hope that such an event will be introduced? 

It's hard to provide my users and contributors with the level of support they deserve when there are no notifications about their questions and ideas.

Thanks!

**Reply by Relequestual** [reply]
I've successfully used Zapier as a bridge using the API. It took me a few hours to get just right, but that's less than it would take me to spin up a service to do the same.

**Reply by folex** [reply]
@Relequestual awesome! Can you share some technical details & struggles you had to overcome? So it's easier for me (and maybe others) to set up.

**Reply by Relequestual** [reply]
@folex Apologies for the delay. Here's a link to where I pasted the ZAP export: https://github.com/github/feedback/discussions/2943#discussioncomment-748127

and the actual link for ease: https://gist.github.com/Relequestual/7684fb3df14d43b7525d9e71d44da65a

**Reply by bribroder** [reply]
Hi @evi-liu would it be possible to update this discussion when the feature is available for testing or just get an update on your timeline? It sounds like you've been actively working on this and it would be great help in encouraging my team to use discussions just to express that activity! https://twitter.com/tweetsbyevi/status/1445095905347002390

**Comment by kyungpyoda** [comment]
This is exactly what I want. 👍

**Comment by erikr** [comment]
I second the desire for the Slack GitHub app to support Discussions.

**Comment by arski** [comment]
+100 and as many as I can! The current issue I see with teams facing when adopting Discussions is that they are in "yet another conversation forum" - whereas if they were where teams already have these same conversations today - Slack, Teams.. - it would be a great integration!

**Comment by akaralar** [comment]
Exactly what we need as well. Discussions go stale when there isn't a constant reminder to comment and conclude them, which could be achieved with the slack github app as it works for PR comments, issue comments etc...

**Comment by folex** [comment]
Same problem here: when people post questions to discussions, I'm always late to reply because I don't have any way of being notified about them. In comparison, on new issues I receive a Slack message and can reply asap.

**Comment by tobybellwood** [comment]
Now that discussions is out of beta, is Slack app integration close?

**Reply by laymonage** [reply]
[Coming soon, it seems.](https://twitter.com/tweetsbyevi/status/1427752653535137796)

**Comment by psifertex** [comment]
I just saw the slack integration is now requesting permissions for read-only access to discussions. Does that mean this is now possible?

**Reply by psifertex** [reply]
Yes! It works! I'm pretty hyped. Not sure why the documentation isn't on the master branch, but the currently deployed plugin worked for me as described on: 

https://github.com/integrations/slack/tree/discussions-readme-update

Just use:

```/github subscribe owner/repo discussions``` to get notification of discussions.

**Reply by piannaf** [reply]
Awesome! I just saw that permission, too

I was able to subscribe with the command you shared 🎉 

However, I tested with an existing integration I have via https://pipedream.com/ that notified me of the topic getting created, pinned, commented, and answered. But, the official github integration did not notify about the pin or the comment

![image](https://user-images.githubusercontent.com/70764/146019896-844978e0-2de8-4970-abc9-f65c9de2363c.png)

**Reply by bribroder** [reply]
Just want to chip in that my team definitely wants notifications on comments as well as new and answered discussions--so nice to see this first step implemented though!

**Reply by JonSEng** [reply]
Hi @evi-liu , I was thankful to see https://github.com/community/community/discussions/2844#discussioncomment-2061814 and wondered if we could add a feature request so that we get notifications on comments as well.
Ideally, we can subscribe to discussions in both ways:
- Github Notifications for Discussions as Slack notifications (Slack alerts that mirror [Github Notifications](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications))
- Being able to subscribe to a Slack channel for all Discussion comments as well - an "activity feed" of sorts.

🙏🏼

**Reply by corneliusroemer** [reply]
Notifications on comments/answers are still missing sorely!
There's a separate discussion for this here: https://github.com/orgs/community/discussions/2844#discussioncomment-6119006

**Comment by ravil-mobile** [comment]
Please, add notifications on comments. It would be so useful!!!

**Reply by osterman** [reply]
@evi-liu since this issue is closed, should a new feature request be opened to add notifications on comments?

**Reply by corneliusroemer** [reply]
See https://github.com/orgs/community/discussions/2844#discussioncomment-6119006

**Comment by evi-liu** [comment, marked as answer]
Thanks for your patience! Discussions are now integrated with Slack! 🎉

You can try it out with `/github subscribe owner/repo discussions`.

You'll receive notifications whenever a discussion is created or when a discussion has a comment [marked as an answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer).

This feature requires granting the Slack app `read` permissions for discussions in your organization. If necessary, you should be prompted to update the app permissions when subscribing to discussions. Please note that you must be an organization owner to approve this permissions update.

Let us know if you have any feedback!

**Reply by prudhomm** [reply]
this is excellent! thanks a lot

**Reply by jordan-brough** [reply]
I'd like to subscribe to just one `category` for a particular repo. Is that possible?

**Reply by philcluff** [reply]
Hey @evi-liu, can you confirm this is available globally? We don't see this option on the GitHub integration in Slack, and don't see any update to the bot pending. Thanks!

**Reply by evi-liu** [reply]
@philcluff Yes, this should be available globally. Did you get an error when you tried the above command in one of your Slack channels?

**Reply by JimmyAppelt** [reply]
@evi-liu as mentioned here https://github.com/github/feedback/discussions/2844#discussioncomment-1852574 notifications for replies/comments also would be awesome to have.

**Reply by OscarPeredo** [reply]
@evi-liu 
Sorry if this has already been asked or mentioned somewhere else but, is this available for Github Enterprise (GHE)?

**Reply by talyh** [reply]
> I'd like to subscribe to just one category for a particular repo. Is that possible?

I was just fiddling with this because I was getting annoyed with the notifications setup

For me `/github subscribe owner discussions` actually subscribed me to all things in our org

`/github unsubscribe owner pulls` unsubbed me from PRs (and then repeated for everything that's not discussions). (this didn't impact the notifications for PRs where I'm tagged, so all good)

It does feel like a bug though, since the first comment did specify a single category and the integration just ignored it and subscribed me to everything

**Comment by sethlivingston** [comment]
Is there a way to have notifications sent to a channel instead of individual Slack users who have issue the `/github subscribe...` command?

**Comment by Dania-Dl** [comment]
Hi, Can we have a `real-time alert` for discussion updates (e.g new reply, comment, mention) as part of  our `Scheduled reminder`

I want to be notified in a direct message, about the discussion that I'm part of, as to the way I have for pull requests.

**Comment by zoshigayan** [comment]
[Get notifications for comment/reply of Discussions in slack (or equivalent) · community · Discussion #56821](https://github.com/orgs/community/discussions/56821)

**Comment by ascheman** [comment]
Is there a Slack Plugin needed (which one)? I get
<img width="295" alt="image" src="https://github.com/community/community/assets/456000/b136de9e-7a1b-4e9f-84db-8a8b69681afc">

**Comment by Tarian300** [comment]
Yes, you can definitely get notifications from discussions in Slack. We had a similar need for keeping track of conversations, and Slack's notifications have been super helpful. You can configure them for specific channels or keywords, and even customize how often you want to get pinged. It really helps streamline responses and keeps everyone in the loop.

When we switched to Slack, the pricing was manageable. The [slack pricing insights](https://www.spendbase.co/vendors/slack/) I found showed that the free plan covers most small teams well, but we opted for the Pro plan for more integrations. They also offer occasional discounts, which helped us save.
